src/game.cpp

```diff
@@ -4534,6 +4534,42 @@ void Game::internalDecayItem(Item* item)
 {
 	const ItemType& it = Item::items[item->getID()];
 	if (it.decayTo != 0) {
+		Player* player = item->getHoldingPlayer();
+		if (player && player->isEquippedItem(item)) {
+			bool needUpdateSkills = false;
+			for (int32_t i = SKILL_FIRST; i <= SKILL_LAST; ++i) {
+				if (it.abilities && it.abilities->skills[i] != 0) {
+					needUpdateSkills = true;
+					player->setVarSkill(static_cast<skills_t>(i), -it.abilities->skills[i]);
+				}
+			}
+
+			if (needUpdateSkills) {
```

Comment:
This update here is an un-needed duplicate. You already update at the end where it should be.